app/controllers/teams_controller.rb

```diff
@@ -39,11 +38,11 @@ def edit
   # POST /teams.xml
   def create
     @team = Team.new(allowed_params)
-
+    authorize @team
     respond_to do |format|
       if verify_recaptcha && ( @team = TeamOperations::Create.(@team, current_user) )
         format.html do
-          session[:team_slug] = @team.slug
+          session[:current_team_slug] = @team.slug
```

Comment:
Have you searched in the whole project for usages of `session[:team_slug]` to replace it?